docs/source/architecture/journal.rst

```diff
@@ -2,476 +2,196 @@
-implied dependencies, with each batch being implicitly dependent on the
-previous.  Any dependencies should be verified when the final set of batches is
-produced.  This would be enforced by the use of strict ordering of the batches
-during execution time.  Future implementations may provide a way to verify
-dependencies across input batches.
+   The BlockStore is expected to be coherent at all times. An error in the
+   BlockStore is considered a non-recoverable error for the validator. Such
+   critical errors would include missing blocks, bad indexes, missing chain
+   reference, incomplete blocks or invalid blocks.
 
-Transaction family authors who need to provide batches that will be included,
-need to provide their own tool to produce GenesisData, with the batches they
-require for the process. Each individual tool may manage their batch and
-transaction dependencies explicitly within the context of their specific
-genesis batches.
 
-Example
-~~~~~~~
+The BlockCache
+==============
 
-The following example configures the validator to use PoET consensus
-and specifies the appropriate settings:
+The BlockCache is an in-memory construct that is rebuilt when the system is
+started. It holds the working set of blocks for the validator.
 
-.. code-block:: bash
+The BlockCache tracks the processing state of each block as valid, invalid, or
+unknown.
 
-        sawset proposal create \
-          -k <signing-key-file> \
-          -o sawset.batch \
-          sawtooth.consensus.algorithm=poet \
-          sawtooth.poet.initial_wait_timer=x \
-          sawtooth.poet.target_wait_time=x \
-          sawtooth.poet.population_estimate_sample_size=x
-	  sawadm genesis \
-          sawset.batch
+- Valid blocks have been proven to be valid by the ChainController. All blocks
+  in the BlockStore are considered valid.
 
-A genesis.batch file will written to the validator's data directory.
+- Invalid blocks have failed validation or have an invalid block as a
+  predecessor.
 
-Block Creation
---------------
+- Unknown blocks have not yet completed validation. Usually, these blocks have
+  just arrived from the Completer.
 
-On startup, the validator would use the resulting genesis.batch file to produce
-a genesis block under the following conditions:
+If a block is not present in the BlockCache, the validator looks in the
+BlockStore for the block. If it is not found or the lookup fails, the block is
+marked as unknown. If the block is found in the BlockStore, it is loaded into
+the BlockCache and marked as valid.
 
-* The genesis.batch file exists
-* There is no block specified as the chain head
+The BlockCache keeps blocks that are currently relevant, tracked by the last
+time the block was accessed. Periodically, blocks that have not been accessed
+recently are purged from the block cache, but only if none of the other blocks
+in the BlockCache reference those blocks as predecessors.
```

Comment:
I am not really sure why we are documenting the BlockCache and BlockStore. These are internal data structures that are in the process of being replaced.

Reply:
But it is accurate.

Reply:
We're documenting it because the picture shows these two items. (And because it was documented before.)  I vote that we leave the information here for now.